### PR TITLE
 Inital step to refactor test/e2e/framework/util.go file, move error handle functions to new package(Initial one)

### PR DESCRIPTION
**Pull request by WanLinghao**
/kind cleanup
 Refactor` test/e2e/framework/util.go` file, move error handle functions to new package.
This patch is just a initial one, it copy the error functions into a new package.And multiple PRs (steps) will follow. After all reference changed, I would delete the original error handle functions in `test/e2e/framework/util.go` file

ref: https://github.com/kubernetes/kubernetes/issues/77095
```release-note
NONE
```

### CODE REVIEW

**Comment by WanLinghao** [comment]
PTAL
/assign @neolit123 @timothysc @andrewsykim

File: test/e2e/framework/error/error.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import (
+	"github.com/onsi/gomega"
+
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+// ExpectNoError checks if "err" is set, and if so, fails assertion while logging the error.
+func ExpectNoError(err error, explain ...interface{}) {
+	ExpectNoErrorWithOffset(1, err, explain...)
+}
```
Comment (neolit123) [inline test/e2e/framework/error/error.go]: the PR doesn't seem to be moving the error utils however, it's only copying them into a new package.
see here:
https://github.com/kubernetes/kubernetes/blob/948ad14f7081a844f6b9072aa93b828e9bf2fcbd/test/e2e/framework/util.go#L2041-L2066

**Review by neolit123 — COMMENTED** [review]
/kind cleanup
/priority backlog

**Comment by spiffxp** [comment]
/hold
Please address @neolit123 's comments.  I'm not sure I buy the need to create a new package here.  It's also unclear to me which part of https://github.com/kubernetes/kubernetes/issues/77095 this is addressing

File: test/e2e/framework/error/error.go (hunk 1 above)
Comment (WanLinghao) [reply test/e2e/framework/error/error.go]: `ExpectNoError` function was widely used in ./test directory. To make it easier to review, I want to do this step by step. I would delete the corresponding functions in kubernetes/test/e2e/framework/util.go after everything being done.

**Comment by WanLinghao** [comment]
@spiffxp Hi, we need to move err functions into its own package like what we already do against log functions(ref:https://github.com/kubernetes/kubernetes/issues/77359)
This is necessary to avoid package import cycle

File: test/e2e/framework/error/error.go (hunk 1 above)
Comment (neolit123) [reply test/e2e/framework/error/error.go]: thanks for the explanation. please update the PR title and description to indicate that this is not a complete move, but rather that an intermediate copy is created and that multiple PRs (steps) will follow.

Comment (WanLinghao) [reply test/e2e/framework/error/error.go]: @neolit123 Updated

**Comment by timothysc** [comment]
So there are legit parts of framework that should be broken apart that will not be used by everyone so having separate library style structure makes sense.  

The error handling though it pretty fundamental and everyone uses.  Would there be a use case for importing framework and not using the error routines?

**Comment by andrewsykim** [comment]
My ideas for the error wrappers were to put it in a `ginkgo_wrappers.go` file but still inside framework package, trying to move any ginkgo wrappers (Skip*, Error*, etc) into their own package won't be feasible (at least right now). https://github.com/kubernetes/kubernetes/pull/77160is the PR to do this

**Comment by WanLinghao** [comment]
@andrewsykim OK, so maybe I could rebase this patch after #77160 merged

**Comment by spiffxp** [comment]
Agree these should be in an area related to ginkgo, just calling it errors seemed way too specific to me

**Comment by timothysc** [comment]
I'm going to close this one in favor of https://github.com/kubernetes/kubernetes/pull/77160.  

@WanLinghao thank you for your PR but we're going to pass on this one.  There is plenty of other low hanging fruit that we could use your help on in the prioritized list linked here (https://docs.google.com/document/d/1TOC8vnmlkWw6HRNHoe5xSv5-qv7LelX6XK3UVCHuwb0/edit#heading=h.ukbaidczvy3r)

**Pull request closed without merging**